### ISSUE TITLE
The IDE compiler have a lot of wrong problem report, maven build is okay

### ISSUE THREAD

**Issue by yiukalun** [spring-projects/sts4#1159]
The IDE compiler wrongly reporting problem for several scenario.

1. Complaining about JpaRepository using composite key. When using composite key in JPA class, extending JpaRepository interface will report error "Expected Domain ID type is 'java.lang.Long", but it only complaining on some class, some other class using same composite key will have no problem.

2. I'm using org.json.JSONObject, the IDE already reference to the spring-starter org.json package and keep complaining that the toMap() function is not available from class.

**Comment by martinlippert** [comment]
Can you attach a sample project that reproduces these wrong problem reporting? That would be awesome.

With regards to 1., this clearly sounds like an issue with the Spring specific validation that we should fix.
With regards to 2., this sounds like a project setup or dependency resolution issue, the sample project would be interesting to analyze here.

Which IDE are you using? The Eclipse based Spring Tools or working inside of VSCode?

**Comment by BoykoAlex** [comment]
@yiukalun any chance you can paste here a code snippet for something that has a problem marker that reads: `Expected Domain ID type is 'java.lang.Long'`?

**Comment by yiukalun** [comment]
![compositeKey](https://github.com/spring-projects/sts4/assets/11970668/b81f17f2-3264-430f-8e9c-289f4363a408)
![compositeKey1](https://github.com/spring-projects/sts4/assets/11970668/d7f5b5d3-aa7a-41b1-966c-26c05a9d7d31)

**Comment by BoykoAlex** [comment]
@yiukalun I have pushed what i think would the fix for the composite repo key: de4a1322cf060cf8e4a7face846376e74805be65
Please try to update your sts from a nightly update site starting from ~2 hours from now. See https://cdn.spring.io/spring-tools/snapshot/STS4/nightly-distributions.html for snapshot distros and update sites

**Comment by martinlippert** [comment]
> 2\. I'm using org.json.JSONObject, the IDE already reference to the spring-starter org.json package and keep complaining that the toMap() function is not available from class.

@yiukalun If you still face this issue and if you can create a small sample project for this, please create a separate issue for this and attach the sample project. Happy to look into this.

**Comment by BoykoAlex** [comment]
@yiukalun should be fixed. Please re-open if you still see issues around composite key. (The fix is de4a1322cf060cf8e4a7face846376e74805be65

**Comment by raneem13j** [comment]
I have the same problem, I am using VS code, and in another repo it works

`package com.Dazzle.app.Dazzle.repositories;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.Dazzle.app.Dazzle.models.user.ERole;
import com.Dazzle.app.Dazzle.models.user.Role;

@Repository
public interface RoleRepository extends JpaRepository<Role, Long> {
  Optional<Role> findByName(ERole name);
}
`

`package com.Dazzle.app.Dazzle.models.user;

import jakarta.persistence.*;

@Entity
@Table(name = "roles")
public class Role {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
  
    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private ERole name;


    public Role() {

    }
  
    public Role(ERole name) {
      this.name = name;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public ERole getName() {
        return name;
    }

    public void setName(ERole name) {
        this.name = name;
    }
    
}
`

**Comment by BoykoAlex** [comment]
@raneem13j I'm not an expert in Spring Data :-\ I've trimmed your use case to the one below which I hope is showing the same issue:
```java
			package demo;
			
			
			import org.springframework.data.jpa.repository.JpaRepository;
			import org.springframework.stereotype.Repository;
			
			@Repository
			public interface RoleRepository extends JpaRepository<Role, Long> {
				
			}
```

and the domain class:
```java
			package demo;
			
			import javax.persistence.Entity;
			import javax.persistence.Id;
			import javax.persistence.Table;
			
			@Entity
			@Table(name = "roles")
			public class Role {
			
				@Id
				private Integer id;
			
				public Integer getId() {
					return id;
				}
			
				public void setId(Integer id) {
					this.id = id;
				}
			
			}

```

It might be right in forcing to switch the repo key to `Inetger` to match the domain class id but as I mentioned earlier I'm far from spring data expert. Thus, I left them a question asking whether the marker is correct in this case. If it is not correct I'm curious what would be the rule for this case... both classes have corresponding primitive types if one primitive type can be cast to another perhaps is the rule (?)

**Comment by BoykoAlex** [comment]
@raneem13j Fixed with 1665b3f51bfa4e2532f0e58ef88bcbe1a770b67d
